ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m

```diff
@@ -51,58 +50,50 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
     NSString *certificateSubject = nil;
     NSData *certificateData = nil;
     NSString *certificateIssuer  = nil;
-    ADAuthenticationError *adError = nil;
-    
-    if (error)
-    {
-        *error = nil;
-    }
+    NSData *issuer = nil;
+    OSStatus status = noErr;
     
     MSID_LOG_VERBOSE(context, @"Attempting to get WPJ registration information");
+    identity = [self copyWPJIdentity:context issuer:&issuer certificateAuthorities:challenge.protectionSpace.distinguishedNames];
     
-    [self copyCertificate:&certificate identity:&identity issuer:&certificateIssuer context:context error:&adError];
-    if (adError)
+    // If there's no identity in the keychain, return nil. adError won't be set if the
```

Comment:
nit: what if we encounter an error copying identity?

Reply:
Lets discuss offline. If we get any error querying the identity , i think we can safely return NULL. If we decide to fill the error object in this case , user will not be able to recover from this state.